File: src/models/character.py
from datetime import datetime

from pydantic import BaseModel, Field


class Character(BaseModel):
    id: str
    world_id: str
    name: str
    portrait_url: str | None = None
    profile: dict = Field(default_factory=dict)
    graph_node_uuid: str | None = None
    entity_type: str = "character"
    tier: str | None = None
    created_at: datetime = Field(default_factory=datetime.now)
    updated_at: datetime = Field(default_factory=datetime.now)


class CreateCharacterRequest(BaseModel):
    name: str = Field(min_length=1)
    portrait_url: str | None = None
    profile: dict | None = None


class UpdateCharacterRequest(BaseModel):
    name: str | None = None
    portrait_url: str | None = None
    profile: dict | None = None
    tier: str | None = None

File: frontend/src/types/character.ts
export interface CharacterBasic {
  id: string
  name?: string | null
  tier?: 'core' | 'supporting' | 'extra' | null
}

export interface CharacterProfile {
  basic?: CharacterBasic
  brief?: string
  detail?: string
  personality?: string
  speech_style?: string
  [key: string]: unknown
}

export interface Character {
  id: string
  world_id: string
  name: string
  portrait_url: string | null
  profile: CharacterProfile
  graph_node_uuid?: string | null
  entity_type?: string
  tier?: 'core' | 'supporting' | 'extra' | null
  created_at: string
  updated_at: string
}

export interface CreateCharacterRequest {
  name: string
  portrait_url?: string | null
  profile?: CharacterProfile | null
}

export interface UpdateCharacterRequest {
  name?: string | null
  portrait_url?: string | null
  profile?: CharacterProfile | null
  tier?: string | null
}
